stdlib/tests/mem/mod.rs

```diff
@@ -5,6 +5,171 @@ use miden_utils_testing::{
     build_expected_perm, felt_slice_to_ints,
 };
 
+#[test]
```

Comment:
Should we add some proptests?

Reply:
I tried to do so, but the main problem is that we can't assert the correctness of the value copied to some new location: in case of the proptest we can't come up with some formula which will represent the value here. But I'm new to the proptest dependency, so maybe I'm missing something.